src/ui/components/Switch/index.js

```diff
@@ -0,0 +1,69 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+import './Switch.scss';
+
+export default class Switch extends React.Component {
+  static propTypes = {
+    browsertheme: PropTypes.string,
+    checked: PropTypes.bool,
+    className: PropTypes.string,
+    disabled: PropTypes.bool,
+    label: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    onChange: PropTypes.func,
+    onClick: PropTypes.func,
+    progress: PropTypes.number,
+    success: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    checked: false,
+    disabled: false,
+    success: false,
+  }
+
+  getProgress() {
+    const { progress } = this.props;
+    if (progress === Infinity) {
+      return 100;
+    } else if (progress === -Infinity) {
+      return 0;
+    }
+    return progress;
+  }
+
+  render() {
+    const {
+      browsertheme, checked, className, disabled, label, name, onChange, onClick, progress,
+      success,
+    } = this.props;
+    const identifier = `install-button-${name}`;
+    const hasProgress = progress !== undefined;
+    const classes = classNames('Switch', className, {
+      'Switch--indefinite': progress === Infinity,
+      'Switch--indefinite-reverse': progress === -Infinity,
+      'Switch--progress': hasProgress && progress !== Infinity && progress !== -Infinity,
+      'Switch--success': success,
+    });
+
+    return (
+      <div className={classes} onClick={onClick}
+        data-progress={hasProgress ? this.getProgress() : 0}>
+        <input
+          id={identifier}
+          className="visually-hidden"
+          checked={checked}
+          disabled={disabled}
+          onChange={onChange}
+          ref={(ref) => { this.themeData = ref; }}
```

Comment:
The ref `this.themeData` looks like it's still accessed by `InstallButton` in `handleClick`. 

More broadly, the approach of using data attributes to know which theme to install seems very un-react to me. Since everything is a react component with its own data already, you shouldn't need data attributes. This makes me think that the install action isn't attached to the right component.